hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkDataSetProcessor.java

```diff
@@ -337,6 +338,26 @@ public Partitioner getPartitioner(DataSet<LocatedRow> dataSet, ExecRow template,
     }
 
     @Override
+    public StructType getExternalFileSchema(String storedAs, String location){
+        StructType schema = null;
+        if (storedAs!=null) {
+            if (storedAs.toLowerCase().equals("p")) {
+                schema =  SpliceSpark.getSession().read().parquet(location).schema();
+
+            }
+            if (storedAs.toLowerCase().equals("o")) {
+                schema =  SpliceSpark.getSession().read().orc(location).schema();
+
+            }
+            if (storedAs.toLowerCase().equals("t")) {
+                schema =  SpliceSpark.getSession().read().csv(location).schema();
+            }
+        }
+
+        return schema;
+    }
+
+    @Override
```

Comment:
Can you create a seperate JIRA for that and I will clean it up. I would rather have that in master first so it is done and work on the clean up later on.